pilot/pkg/secrets/kube/multicluster.go

```diff
@@ -48,19 +50,18 @@ func NewMulticluster(client kube.Client, localCluster, secretNamespace string) *
 		func(k string) error { m.deleteMemberCluster(k); return nil },
 		secretNamespace,
 		time.Millisecond*100,
-		make(chan struct{}))
+		stop)
 	m.secretController = sc
 	return m
 }
 
 func (m *Multicluster) addMemberCluster(clients kube.Client, key string) {
-	stopCh := make(chan struct{})
 	log.Infof("initializing Kubernetes credential reader for cluster %v", key)
 	sc := NewSecretsController(clients, key)
 	m.m.Lock()
 	m.remoteKubeControllers[key] = sc
 	m.m.Unlock()
-	clients.RunAndWait(stopCh)
+	clients.RunAndWait(m.stop)
```

Comment:
This might deserve a bigger refactor. I think there are 2 issues here:
1. Sharing the stop channel across all clusters doesn't let us remove a cluster from a running istiod (not sure this is the only thing stopping this, but the ability to do that is attractive to users with many-cluster meshes)
2. We call `RunAndWait` both from the serviceregistry controller and this secrets controller. 

I think the remote secret controller (not to be confused with the one I'm commenting on here) should be in charge of  managing the `stopCh` and passing it to the `addCallback`, then calling `RunAndWait` after the callback. This avoids distributing too much of the add/remove cluster logic WRT kube client details. 

https://github.com/istio/istio/blob/e50de12afd9613d59462a008a27ef94f1e4ed652/pkg/kube/secretcontroller/secretcontroller.go#L301

Reply:
Btw, not saying that this PR creates a new issue or makes anything worse.

Reply:
Agree, the stop channel should be separate. In the long run, there should be only one multicluster.